.github/workflows/dry-run.yml

```diff
@@ -9,5 +9,21 @@ on:
   merge_group:
 
 jobs:
-  dry-run:
-    uses: cloudscape-design/actions/.github/workflows/dry-run.yml@main
```

Comment:
Would it be possible to parametrise the dry-run instead and create some sort of dependency mapping between the project? The idea is that in every project we still call the dry-run script but provide the project name like "board-components" or "collection-hooks" and the script then decides which projects to build.

Reply:
Thinking out loud but would that not complicate the [workflow](https://github.com/cloudscape-design/actions/blob/main/.github/workflows/dry-run.yml) quite a bit? Would it end up being if statements everywhere and then what if a later build `needs` a build that was never run and so it cannot run? I don't think that `needs` list can be easily dynamically built up. 

I tried a [dynamic workflow](https://github.com/cloudscape-design/test-utils/actions/runs/11288423575/attempts/17) (you can see how many attempts I tried with this 1 😅) before this and the additional component tests added a lot of complexity to making a dynamic dry-run. Github actions is still fairly limited in this regard. 

I settled on the solution of:
- When building from the bottom of the tree up its about 3 minutes to build everything before components its about 1 minute to build each level before components. So optimising a level of build is ~1 minute saved. E.g Building from test-utils up 
- The greatest time saved was at this point where building from the bottom up until the end of testing components was 10 minutes and so changing board-components, chat-components and code-view to a custom dry-run introduced no extra complexity and the custom code needed is quite minimal (as seen in this PR). 
- I have a script that can generate the build tree from a package. Ideally I'd like to use that somehow but not sure yet where

![image](https://github.com/user-attachments/assets/c05ab86e-bd47-4f3a-91ff-332c9afe3de3)

Reply:
The [second option](https://github.com/cloudscape-design/actions/pull/60) I briefly looked at was having this in the actions repo but given that these are not reusable by any other package I opted for having them in the individual repos

Reply:
If we can generate a build tree from a package I wonder if we can use that script to generate the workflow file, too. This can replace the dry-run job in actions, maybe. Say, we can run the script as a post-install in each package.

However, I think the current changes are still good and can be merged as is - I am just sharing some thoughts as to how we can possibly improve the workflows for long-term.

Reply:
Possibly an addition for the [build-tools](https://github.com/cloudscape-design/build-tools) package :)

Reply:
Found a [new path forward](https://github.com/cloudscape-design/actions/pull/63) that seems to work. I will revert this PR in favour of this approach.